File: services/backend-laravel/app/Models/QuestionnaireAttempt.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class QuestionnaireAttempt extends Model
{
    protected $fillable = [
        'user_id',
        'questionnaire_version',
        'bank_version',
        'attempt_version',
        'answers_count',
        'total_count',
        'answers_json',
        'profile_json',
        'summary_json',
        'result_label',
        'result_highlights_json',
        'completed_at',
    ];

    protected function casts(): array
    {
        return [
            'answers_count' => 'integer',
            'total_count' => 'integer',
            'answers_json' => 'array',
            'profile_json' => 'array',
            'summary_json' => 'array',
            'result_highlights_json' => 'array',
            'completed_at' => 'datetime',
        ];
    }

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }
}

File: apps/flutter_elitesync_module/lib/features/questionnaire/domain/entities/questionnaire_attempt.dart
class QuestionnaireAttempt {
  const QuestionnaireAttempt({
    required this.id,
    required this.questionnaireVersion,
    required this.bankVersion,
    required this.attemptVersion,
    required this.answersCount,
    required this.totalCount,
    required this.resultLabel,
    required this.resultHighlights,
    required this.profileComplete,
    required this.completedAt,
  });

  final int id;
  final String questionnaireVersion;
  final String bankVersion;
  final String attemptVersion;
  final int answersCount;
  final int totalCount;
  final String resultLabel;
  final List<String> resultHighlights;
  final bool profileComplete;
  final DateTime? completedAt;
}
